src/libraries/System.Private.CoreLib/src/System/Text/RunePosition.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using System.Buffers;
+
+namespace System.Text;
+
+/// <summary>
+/// The position in unicode data that allows deeper data inspection.
+/// </summary>
+/// <param name="Rune">
+/// Unicode scalar value <see cref="System.Text.Rune"/> of the current char in unicode data.
+/// Invalid encoded unicode char will be represented by <see cref="Rune.ReplacementChar"/> value.
+/// </param>
+/// <param name="StartIndex">The index of current char in unicode data.</param>
+/// <param name="Length">The length of current char in unicode data.</param>
+/// <param name="WasReplaced">
+/// true if current char is invalid encoded and <see cref="Rune"/> was replaced by <see cref="System.Text.Rune.ReplacementChar"/>.
+/// false it current char is correct encoded and <see cref="Rune"/> contain its scalar value.
+/// </param>
+/// <remarks>
+/// Invalid unicode char will be represented in <see cref="Rune"/> by <see cref="Rune.ReplacementChar"/>.
+/// </remarks>
+public readonly record struct RunePosition(Rune Rune, int StartIndex, int Length, bool WasReplaced)
```

Comment:
`records` come with some additional footprint, so (at least now) they are avoided in CoreLib.
Use a regular readonly struct instead.